[yaml-cpp] Cherry-pick of https://github.com/jbeder/yaml-cpp/pull/1336

We can drop this patch once we upgrade to a release that contains it.

From: Jeremy Nimmer <jeremy.nimmer@tri.global>
Date: Mon, 30 Dec 2024 07:23:23 -0800
Subject: [PATCH] Avoid including <iostream> in library code

Including iostream means introducing the static (global) constructors
and destructors for std::cin, std::cerr, and std::cout. That extra
init and fini code is undesirable when those streams are not actually
used.

Instead, we'll use the narrower includes for exactly what's needed,
i.e., <istream> or <ostream>.

--- src/indentation.h
+++ src/indentation.h
@@ -7,7 +7,6 @@
 #pragma once
 #endif
 
-#include <iostream>
 #include <cstddef>
 
 #include "yaml-cpp/ostream_wrapper.h"

--- src/ostream_wrapper.cpp
+++ src/ostream_wrapper.cpp
@@ -2,7 +2,7 @@
 
 #include <algorithm>
 #include <cstring>
-#include <iostream>
+#include <ostream>
 
 namespace YAML {
 ostream_wrapper::ostream_wrapper()

--- src/stream.cpp
+++ src/stream.cpp
@@ -1,4 +1,4 @@
-#include <iostream>
+#include <istream>
 
 #include "stream.h"
 

--- src/stream.h
+++ src/stream.h
@@ -11,7 +11,7 @@
 #include <cstddef>
 #include <deque>
 #include <ios>
-#include <iostream>
+#include <istream>
 #include <set>
 #include <string>
 

--- src/token.h
+++ src/token.h
@@ -8,7 +8,7 @@
 #endif
 
 #include "yaml-cpp/mark.h"
-#include <iostream>
+#include <ostream>
 #include <string>
 #include <vector>
 
